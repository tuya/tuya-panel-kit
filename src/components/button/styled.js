import React from 'react';
import { View, TouchableOpacity } from 'react-native';
import styled, { css } from 'styled-components/native';
import TYText from '../TYText';
import IconFont from '../iconfont';
import { defaultTheme } from '../theme';
import { RatioUtils, ThemeUtils } from '../../utils';

const { convertX: cx } = RatioUtils;
const { parseToCss, getTheme, ThemeConsumer } = ThemeUtils;

const DEFAULT_THEME = defaultTheme.button.light;

export const SIZE_MAP = {
  small: cx(32),
  normal: cx(40),
  large: cx(48),
};

const BTN_TEXT_STYLES_MAP = {
  left: css`
    font-size: 16;
    margin-right: 8px;
  `,
  right: css`
    font-size: 16;
    margin-left: 8px;
  `,
  top: css`
    margin-bottom: 4px;
  `,
  bottom: css`
    margin-top: 4px;
  `,
};

const sizeStyles = props => {
  const bgWidth = getTheme(props, 'button.bgWidth');
  const bgHeight = getTheme(props, 'button.bgHeight');
  const bgRadius = getTheme(props, 'button.bgRadius');
  if (props.size === 'noSet') {
    const styles = [];
    if (typeof bgWidth === 'number') styles.push(`width: ${bgWidth}`);
    if (typeof bgHeight === 'number') styles.push(`height: ${bgHeight}`);
    if (typeof bgRadius === 'number') styles.push(`border-radius: ${bgRadius}`);
    if (styles.length === 0) return null;
    const stylesStr = styles.join('\n');
    return css`
      ${stylesStr}
    `;
  }
  const size = typeof props.size === 'number' ? props.size : SIZE_MAP[props.size];
  // TODO: 确认是否能直接将按钮背景改为圆角
  return css`
    width: ${bgWidth || size}px;
    height: ${bgHeight || size}px;
    border-radius: ${bgRadius || size / 2}px;
  `;
};

const bgStyles = props => {
  if (props.type !== 'primary') {
    return css`
      background-color: transparent;
    `;
  }
  return css`
    background-color: ${getTheme(props, 'button.bgColor', DEFAULT_THEME.bgColor)};
  `;
};

const btnTextStyles = props => {
  if (!props.textDirection) {
    return null;
  }
  return BTN_TEXT_STYLES_MAP[props.textDirection] || BTN_TEXT_STYLES_MAP.right;
};

export const StyledBtnWrapper = styled(View)`
  align-self: ${props => (props.stretch ? 'stretch' : 'center')};
  align-items: center;
  justify-content: center;
  ${props => {
    const margin = getTheme(props, 'button.margin', DEFAULT_THEME.margin);
    return parseToCss(margin, 'margin');
  }};
`;

export const StyledBtnContainer = styled(TouchableOpacity)`
  align-self: ${props => (props.stretch ? 'stretch' : 'center')};
  align-items: center;
  justify-content: center;
`;

export const StyledBtn = styled(View)`
  align-items: center;
  justify-content: center;
  overflow: hidden;
  ${props => sizeStyles(props)};
  ${props => bgStyles(props)};
`;

export const StyledBtnText = styled(TYText)`
  text-align: center;
  font-size: ${props => getTheme(props, 'button.fontSize', DEFAULT_THEME.fontSize)};
  opacity: ${props => (props.disabled ? 0.5 : 1)};
  color: ${props => getTheme(props, 'button.fontColor', DEFAULT_THEME.fontColor)};
  ${props => btnTextStyles(props)};
`;

export const StyledBadge = styled(View)`
  position: absolute;
  border-radius: ${cx(8)}px;
  padding: ${`2px ${cx(8)}px`};
`;

export const StyledBadgeText = styled(TYText)`
  font-size: 10px;
  text-align: center;
`;

export const StyledIconFont = props => {
  // eslint-disable-next-line react/prop-types
  const { size, color, ...rest } = props;
  return (
    <ThemeConsumer>
      {theme => {
        const propsWithTheme = { ...props, theme };
        return (
          <IconFont
            size={size || 28}
            color={color || getTheme(propsWithTheme, 'button.iconColor', DEFAULT_THEME.iconColor)}
            {...rest}
          />
        );
      }}
    </ThemeConsumer>
  );
};
