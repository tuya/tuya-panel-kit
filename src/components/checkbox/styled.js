import { TouchableOpacity } from 'react-native';
import styled from 'styled-components/native';
import TYText from '../TYText';
import IconFont from '../iconfont';
import { defaultTheme } from '../theme';
import { RatioUtils, ThemeUtils } from '../../utils';

const { convertX: cx } = RatioUtils;
const { getTheme } = ThemeUtils;

const DEFAULT_THEME = defaultTheme.checkbox.light;

export const StyledCheckbox = styled(TouchableOpacity)`
  flex-direction: row;
  align-self: stretch;
  align-items: center;
  justify-content: flex-start;
  background-color: transparent;
`;

export const StyledIcon = styled(IconFont).attrs({
  size: props => props.size || getTheme(props, 'checkbox.size', DEFAULT_THEME.size),
  color: props => {
    let color;
    if (props.disabled) {
      color =
        props.disabledColor ||
        getTheme(props, 'checkbox.disabledColor', DEFAULT_THEME.disabledColor);
    } else {
      color =
        props.activeColor || getTheme(props, 'checkbox.activeColor', DEFAULT_THEME.activeColor);
    }
    return color;
  },
})``;

export const StyledText = styled(TYText).attrs({
  type: 'title',
  size: 'small',
})`
  margin-left: ${cx(6)}px;
  color: ${props => getTheme(props, 'checkbox.fontColor', DEFAULT_THEME.fontColor)};
`;
