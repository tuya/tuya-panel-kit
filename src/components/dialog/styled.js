import { View, TouchableOpacity, TextInput, StyleSheet } from 'react-native';
import styled from 'styled-components/native';
import { RatioUtils } from '../../utils';
import TYFlatList from '../TYLists/list';
import TYText from '../TYText';
import { dialog } from '../theme/theme-get';

const { convertX: cx } = RatioUtils;

const {
  width,
  radius,
  bgColor,
  lineColor,
  cellHeight,
  titleFontSize,
  titleFontColor,
  subTitleFontSize,
  subTitleFontColor,
  cancelFontSize,
  cancelFontColor,
  confirmFontSize,
  confirmFontColor,
  prompt,
  pressColor,
} = dialog;

/**
 * Common Dialog Variants
 */
export const StyledContainer = styled(View)`
  width: ${width}px;
  align-self: center;
  align-items: center;
  justify-content: center;
  border-radius: ${radius};
  overflow: hidden;
  background-color: ${bgColor};
`;

export const StyledHeader = styled(View)`
  align-self: stretch;
  align-items: center;
  justify-content: center;
  border-bottom-width: ${StyleSheet.hairlineWidth};
  border-bottom-color: ${lineColor};
  height: ${cellHeight}px;
`;

export const StyledContent = styled(View)`
  padding: ${`${cx(32)}px ${cx(32)}px`};
  align-self: stretch;
  align-items: center;
  justify-content: center;
`;

export const StyledFooter = styled(View)`
  align-self: stretch;
  flex-direction: row;
  border-top-width: ${StyleSheet.hairlineWidth};
  border-top-color: ${lineColor};
  height: ${cellHeight}px;
`;

export const StyledTitle = styled(TYText)`
  font-weight: 500;
  font-size: ${titleFontSize}px;
  text-align: center;
  color: ${titleFontColor};
  flex-wrap: wrap;
`;

export const StyledSubTitle = styled(TYText)`
  font-size: ${subTitleFontSize}px;
  text-align: center;
  color: ${subTitleFontColor};
  flex-wrap: wrap;
`;

export const StyledConfirmButton = styled(TouchableOpacity).attrs({
  activeOpacity: 0.8,
})`
  flex: 1;
  align-items: center;
  justify-content: center;
  border-right-width: ${props => (props.bordered ? StyleSheet.hairlineWidth : 0)};
  border-right-color: ${lineColor};
  background-color: ${props => (props.pressActive ? pressColor : bgColor)};
`;

export const StyledCancelButton = styled(TouchableOpacity).attrs({
  activeOpacity: 0.8,
})`
  flex: 1;
  align-items: center;
  justify-content: center;
  border-right-width: ${props => (props.bordered ? StyleSheet.hairlineWidth : 0)};
  border-right-color: ${lineColor};
  background-color: ${props => (props.pressActive ? pressColor : bgColor)};
`;

export const StyledButton = styled(TouchableOpacity).attrs({
  activeOpacity: 0.8,
})`
  flex: 1;
  align-items: center;
  justify-content: center;
  border-right-width: ${props => (props.bordered ? StyleSheet.hairlineWidth : 0)};
  border-right-color: ${lineColor};
`;

export const StyledCancelText = styled(TYText)`
  font-size: ${cancelFontSize}px;
  text-align: center;
  color: ${cancelFontColor};
`;

export const StyledConfirmText = styled(TYText)`
  font-weight: bold;
  font-size: ${confirmFontSize}px;
  text-align: center;
  color: ${confirmFontColor};
`;

/**
 * Prompt Variants
 */

export const StyledInputContainer = styled(View)`
  justify-content: center;
  align-self: stretch;
  margin-top: 28px;
  padding: ${prompt.padding};
`;

export const StyledInput = styled(TextInput).attrs({
  placeholderTextColor: prompt.placeholder,
  underlineColorAndroid: 'transparent',
})`
  font-size: 16;
  color: ${subTitleFontColor};
  padding: 0px;
`;

/**
 * Checkbox variants
 */
export const StyledCheckboxList = styled(TYFlatList)`
  align-self: stretch;
  background-color: ${bgColor};
`;

/**
 * List variants
 */
export const StyledList = styled(TYFlatList)`
  margin-top: ${cx(16)}px;
  align-self: stretch;
  background-color: ${bgColor};
  border-top-color: ${lineColor};
  border-top-width: ${StyleSheet.hairlineWidth};
  border-bottom-left-radius: ${radius};
  border-bottom-right-radius: ${radius};
`;
