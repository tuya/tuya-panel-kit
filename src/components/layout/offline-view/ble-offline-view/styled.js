/* eslint-disable import/prefer-default-export */
import { Text } from 'react-native';
import styled from 'styled-components/native';
import { dialog } from '../../../theme/theme-get';
import TYText from '../../../TYText';

const { titleFontColor, confirmFontColor, confirmFontSize } = dialog;

export const StyledTitle = styled(Text)`
  text-align: center;
  color: ${titleFontColor};
`;

export const StyledCancelText = styled(TYText)`
  text-align: center;
  color: ${confirmFontColor};
  font-size: ${confirmFontSize}px;
`;
