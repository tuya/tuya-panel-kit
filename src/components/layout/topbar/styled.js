import React from 'react';
import { View, Image, TouchableOpacity } from 'react-native';
import styled from 'styled-components/native';
import TYText from '../../TYText';
import IconFont from '../../iconfont';
import { defaultTheme } from '../../theme';
import { ThemeUtils, RatioUtils } from '../../../utils';

const { getTheme, ThemeConsumer } = ThemeUtils;
const { isIos, isIphoneX, statusBarHeight } = RatioUtils;

const DEFAULT_THEME = defaultTheme.topbar.light;

const ALIGN_ITEMS_MAP = {
  left: 'flex-start',
  center: 'center',
  right: 'flex-end',
};

export const TOPBAR_MARGIN = 6;
export const TOPBAR_HEIGHT = isIos ? (isIphoneX ? 88 : 64) : 56;
export const TOPBAR_ACTION_WIDTH = 17;
export const TOPBAR_ACTION_TEXT_WIDTH = 78;

export const StyledTopBarContainer = styled(View)`
  align-self: stretch;
  height: ${TOPBAR_HEIGHT}px;
`;

export const StyledTopBar = styled(View)`
  flex-direction: row;
  align-self: stretch;
  align-items: center;
  justify-content: space-between;
  height: ${isIos ? TOPBAR_HEIGHT - statusBarHeight : TOPBAR_HEIGHT};
  margin-top: ${isIos ? statusBarHeight : 0};
`;

export const StyledTopBarContent = styled(View)`
  position: absolute;
  height: 100%;
  align-self: center;
  align-items: ${props => ALIGN_ITEMS_MAP[props.align] || 'center'};
  justify-content: center;
`;

export const StyledTopBarTitle = styled(TYText).attrs({
  type: 'title',
  size: 17,
})`
  font-weight: 500;
  color: ${props => props.color || getTheme(props, 'topbar.color', DEFAULT_THEME.color)};
`;

export const StyledTopBarSubTitle = styled(TYText).attrs({
  type: 'paragraph',
  size: 'normal',
})`
  color: ${props => props.color || getTheme(props, 'topbar.color', DEFAULT_THEME.color)};
`;

export const StyledTopBarAction = styled(TouchableOpacity)`
  width: ${TOPBAR_ACTION_WIDTH}px;
  height: 100%;
  position: absolute;
  align-items: center;
  justify-content: center;
`;

export const StyledTopBarText = styled(TYText)`
  font-size: 16px;
  color: ${props => props.color || getTheme(props, 'topbar.color', DEFAULT_THEME.color)};
`;

export const StyledImage = props => {
  const { style, ...rest } = props;
  return (
    <ThemeConsumer>
      {theme => {
        const propsWithTheme = { ...props, theme };
        const imageStyle = [
          {
            tintColor: props.color || getTheme(propsWithTheme, 'topbar.color', DEFAULT_THEME.color),
          },
          style,
        ];
        return <Image style={imageStyle} {...rest} />;
      }}
    </ThemeConsumer>
  );
};

export const StyledIconFont = props => {
  const { color, ...rest } = props;
  return (
    <ThemeConsumer>
      {theme => {
        const propsWithTheme = { ...props, theme };
        return (
          <IconFont
            color={color || getTheme(propsWithTheme, 'topbar.color', DEFAULT_THEME.color)}
            {...rest}
          />
        );
      }}
    </ThemeConsumer>
  );
};
