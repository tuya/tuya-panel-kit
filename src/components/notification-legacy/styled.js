import { View, TouchableOpacity } from 'react-native';
import styled from 'styled-components/native';
import { RatioUtils } from '../../utils';
import TopBar from '../layout/topbar';
import TYText from '../TYText';

const { convertX: cx } = RatioUtils;

export const StyledNotification = styled(TouchableOpacity)`
  align-self: stretch;
  align-items: center;
  justify-content: space-between;
`;

export const StyledNotificationContent = styled(View)`
  width: ${cx(351)}px;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  margin: ${`${TopBar.height + 12}px 0px 0px`};
  padding: ${`6px ${cx(16)}px`};
  min-height: 56px;
  border-radius: 28;
  background-color: ${props => props.background};
`;

export const StyledTitle = styled(TYText)`
  flex: 1;
  font-size: 14;
  line-height: 20;
  color: ${props => props.color};
  margin: ${`2px ${cx(8)}px`};
`;

export const StyledButton = styled(TouchableOpacity)`
  width: ${cx(24)}px;
  height: ${cx(24)}px;
  align-items: center;
  justify-content: center;
`;
