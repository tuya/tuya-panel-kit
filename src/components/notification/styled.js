import { View, TouchableOpacity, Image, Text } from 'react-native';
import styled from 'styled-components/native';
import { RatioUtils } from '../../utils';
import TYText from '../TYText';
import IconFont from '../iconfont';

const { viewWidth, convertX: cx } = RatioUtils;

export const StyledNotification = styled(TouchableOpacity)`
  align-self: stretch;
  align-items: center;
  justify-content: space-between;
`;

export const StyledNotificationContent = styled(View)`
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  margin: ${`${12}px 0px ${2}px 0px`};
  border-radius: 22;
  background-color: ${props => props.background};
`;

export const StyledNotificationContainer = styled(View)`
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  margin: ${`10px ${cx(16)}px`};
  min-height: 10;
  background-color: ${props => props.background};
`;

export const StyledTitle = styled(TYText)`
  min-width: ${cx(44)}px;
  max-width: ${viewWidth - 142}px;
  font-size: 14;
  color: ${props => props.color};
  margin: ${`0px ${cx(30)}px 0px ${cx(32)}px`};
`;

export const StyledIconFont = styled(IconFont)`
  position: absolute;
  top: ${cx(3)}px;
`;

export const StyledImage = styled(Image)`
  position: absolute;
  top: ${cx(2)}px;
  height: ${cx(21)}px;
  width: ${cx(24)}px;
`;
