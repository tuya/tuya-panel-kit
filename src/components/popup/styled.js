import React from 'react';
import { View, TouchableOpacity, StyleSheet } from 'react-native';
import styled from 'styled-components/native';
import { RatioUtils, ThemeUtils } from '../../utils';
import Button from '../button';
import TYText from '../TYText';
import Slider from '../slider';
import IconFont from '../iconfont';
import TimerPicker from '../timer-picker';
import TYFlatList from '../TYLists/list';
import DatePicker from '../date-picker';
import SwitchButton from '../switch-button';
import { defaultTheme } from '../theme';
import { popup } from '../theme/theme-get';

const DEFAULT_LIST_THEME = defaultTheme.popup.basic;
const DEFAULT_PICKER_THEME = defaultTheme.picker.light;

export const backIcon =
  'M770.673778 21.959111a56.888889 56.888889 0 0 1 0 80.440889l-402.204445 402.318222 402.204445 402.204445a56.888889 56.888889 0 0 1-80.440889 80.497777L247.751111 544.938667a56.888889 56.888889 0 0 1 0-80.497778L690.232889 21.959111a56.888889 56.888889 0 0 1 80.440889 0z';

const { convertX: cx, isIphoneX, isIos } = RatioUtils;
const { getTheme, ThemeConsumer } = ThemeUtils;
const {
  cellHeight,
  cellBg,
  cellFontSize,
  cellFontColor,
  titleRadius,
  titleHeight,
  titleBg,
  footerRadius,
  bottomBg,
  lineColor,
  titleFontSize,
  titleFontColor,
  cancelFontSize,
  cancelFontColor,
  confirmFontSize,
  confirmFontColor,
  subTitleFontColor,
  backIconColor,
  pressColor,
} = popup;

/**
 * Common Popup
 */
export const Row = styled(View)`
  flex-direction: row;
  align-items: center;
  justify-content: center;
  height: ${cellHeight}px;
  background-color: ${cellBg};
`;

// 为了处理安卓 Modal 在内容框内同时双击会出现Modal关闭问题
export const StyledContainer = styled(TouchableOpacity).attrs({
  activeOpacity: 1,
})`
  /* bottom: -100%; */
  border-top-left-radius: ${titleRadius};
  border-top-right-radius: ${titleRadius};
  border-bottom-left-radius: ${footerRadius};
  border-bottom-right-radius: ${footerRadius};
  background-color: ${bottomBg};
`;

export const StyledTitle = styled(Row)`
  justify-content: space-around;
  height: ${titleHeight}px;
  padding: 0 16px;
  background-color: ${titleBg};
  border-top-left-radius: ${titleRadius};
  border-top-right-radius: ${titleRadius};
`;

export const StyledTitleText = styled(TYText)`
  font-size: ${titleFontSize};
  color: ${titleFontColor};
`;

export const StyledSwitch = styled(SwitchButton).attrs({
  tintColor: props => getTheme(props, 'popup.tintColor', '#e5e5e5'),
})``;

export const StyledCheckout = styled(View).attrs({})`
  width: 18px;
  height: 18px;
  border-radius: 9px;
  border-width: 1px;
  align-items: center;
  justify-content: center;
  border-color: ${props => (props.active ? confirmFontColor : '#e5e5e5')};
  background-color: ${props => (props.active ? confirmFontColor : 'transparent')};
`;

export const StyledFooter = styled(Row)`
  padding-bottom: ${isIphoneX ? 20 : 0};
  margin-top: ${cx(6)}px;
  border-bottom-left-radius: ${footerRadius};
  border-bottom-right-radius: ${footerRadius};
  height: ${props => {
    const height = cellHeight(props);
    return isIphoneX ? height + 20 : height;
  }}px;
`;

export const StyledConfirmButton = styled(TouchableOpacity).attrs({
  activeOpacity: 0.8,
})`
  flex: 1;
  padding: 12px 0;
  align-self: stretch;
  align-items: center;
  justify-content: center;
  border-right-width: ${props => (props.bordered ? StyleSheet.hairlineWidth : 0)};
  border-right-color: ${lineColor};
  background-color: ${props => (props.pressActive ? pressColor : cellBg)};
`;

export const StyledCancelButton = styled(TouchableOpacity).attrs({
  activeOpacity: 0.8,
})`
  flex: 1;
  padding: 12px 0;
  align-self: stretch;
  align-items: center;
  justify-content: center;
  border-right-width: ${props => (props.bordered ? StyleSheet.hairlineWidth : 0)};
  border-right-color: ${lineColor};
  background-color: ${props => (props.pressActive ? pressColor : cellBg)};
`;

export const StyledCancelText = styled(TYText)`
  font-size: ${cancelFontSize};
  color: ${cancelFontColor};
`;

export const StyledConfirmText = styled(TYText)`
  font-weight: bold;
  font-size: ${confirmFontSize};
  color: ${confirmFontColor};
`;

export const StyledSubTitleText = styled(TYText)`
  font-size: ${props => {
    const size = getTheme(props, 'popup.titleFontSize', DEFAULT_LIST_THEME.titleFontSize);
    return (size * 6) / 7;
  }};
  color: ${subTitleFontColor};
  margin-top: ${cx(5)}px;
`;

export const StyledTouchView = styled(TouchableOpacity).attrs({
  activeOpacity: 0.7,
})`
  flex-direction: row;
  align-items: center;
  justify-content: center;
  position: absolute;
  left: 12px;
`;

export const StyledBackText = styled(TYText).attrs({
  color: backIconColor,
  size: titleFontSize,
})`
  margin-left: 2px;
`;

export const StyledBackIcon = props => {
  /* eslint-disable react/prop-types */
  const { color, ...rest } = props;
  return (
    <ThemeConsumer>
      {theme => {
        const propsWithTheme = { ...props, theme };
        return (
          <IconFont
            size={cx(18)}
            color={
              color ||
              getTheme(propsWithTheme, 'popup.backIconColor', DEFAULT_LIST_THEME.backIconColor)
            }
            {...rest}
          />
        );
      }}
    </ThemeConsumer>
  );
};
/**
 * Popup.picker
 */
export const StyledPickerContainer = styled(View)`
  flex-direction: row;
  align-items: center;
  justify-content: center;
  background-color: ${cellBg};
  opacity: ${props => (props.disabled ? 0.6 : 1)};
`;

export const StyledPickerUnit = styled(View)`
  position: absolute;
  top: 0;
  align-items: center;
  justify-content: center;
`;

export const StyledPickerUnitText = styled(TYText)`
  font-size: ${props => getTheme(props, 'picker.unitFontSize', DEFAULT_PICKER_THEME.unitFontSize)};
  font-weight: bold;
  color: ${props => props.pickerUnitColor || cellFontColor};
`;

/**
 * Popup.countdown
 */
export const StyledCountdownContainer = styled(View)`
  flex-direction: row;
  align-items: center;
  padding: 1px 0;
  background-color: ${cellBg};
`;

export const StyledCountdownContent = styled(View)`
  flex-direction: row;
  align-items: center;
  margin: 0px 30px;
  background-color: ${cellBg};
`;

export const StyledOverview = styled(View)`
  flex-direction: row;
  align-items: center;
  overflow: hidden;
`;

/**
 * Popup.timerPicker
 */
export const StyledTimerPickerContainer = styled(View)`
  height: 300px;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  padding: 0 24px;
`;

export const StyledTimerPickerRow = styled(View)`
  flex: 1;
  flex-direction: row;
`;

export const StyledSymbolText = styled(TYText)`
  padding: 0 18px;
  font-size: ${props => getTheme(props, 'picker.unitFontSize', DEFAULT_PICKER_THEME.unitFontSize)};
  color: ${props => getTheme(props, 'picker.unitFontColor', DEFAULT_PICKER_THEME.unitFontColor)};
`;

export const StyledTimerText = styled(TYText)`
  flex: 1;
  align-items: center;
  justify-content: center;
  font-size: 12;
  text-align: center;
  color: ${cellFontColor};
  background-color: ${cellBg};
`;

export const StyledTimerPicker = props => {
  return (
    <ThemeConsumer>
      {globalTheme => {
        const timerPickerTheme = { ...props, theme: globalTheme };
        const fontColor = getTheme(timerPickerTheme, 'popup.cellFontColor');
        let timerStyle;
        /* eslint-disable react/prop-types */
        if (props.style && props.style.backgroundColor) {
          timerStyle = props.style;
        } else if (props.style) {
          timerStyle = {
            ...props.style,
            backgroundColor: getTheme(timerPickerTheme, 'popup.cellBg'),
          };
        } else {
          timerStyle = { backgroundColor: getTheme(timerPickerTheme, 'popup.cellBg') };
        }
        return (
          <View
            style={{
              backgroundColor: getTheme(timerPickerTheme, 'popup.cellBg'),
            }}
          >
            {props.startTitle && props.endTitle && (
              <View
                style={[
                  {
                    flexDirection: 'row',
                    justifyContent: 'space-between',
                    paddingHorizontal: cx(30),
                    position: 'absolute',
                    top: cx(26),
                    zIndex: 1,
                    backgroundColor: getTheme(timerPickerTheme, 'popup.cellBg'),
                    opacity: props.disabled ? 0.6 : 1,
                  },
                ]}
              >
                <StyledTimerText style={{ marginLeft: cx(15) }}>{props.startTitle}</StyledTimerText>
                <StyledTimerText style={{ marginRight: cx(13) }}>{props.endTitle}</StyledTimerText>
              </View>
            )}
            <TimerPicker
              {...props}
              pickerFontColor={fontColor}
              style={[
                isIos && { height: cx(216), marginTop: cx(36) },
                !isIos && { height: cx(216), marginTop: cx(57), marginBottom: cx(26) },
                timerStyle,
              ]}
            />
          </View>
        );
      }}
    </ThemeConsumer>
  );
};

/**
 * Popup.numberSelector
 */
export const StyledSliderContent = styled(View)`
  padding: 32px 0px;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  background-color: ${cellBg};
`;

export const StyledSliderContainer = styled(View)`
  height: 56px;
  margin-top: 16px;
  flex-direction: ${props => props.flexDirection || 'row'};
  align-items: center;
  justify-content: center;
`;

export const StyledSlider = styled(Slider).attrs({
  minimumTrackTintColor: '#0B7CFF',
  maximumTrackTintColor: props => getTheme(props, 'popup.numberSelector.maximumTrackTintColor'),
})`
  width: ${cx(220)}px;
  margin: 0 12px;
`;

export const StyledSliderBtn = styled(Button).attrs({
  iconSize: 26,
  iconColor: props => getTheme(props, 'popup.numberSelector.cellPlusColor'),
})`
  opacity: ${props => (props.disabled ? 0.6 : 1)};
`;

export const StyledDisplayText = styled(TYText).attrs({
  numberOfLines: 1,
})`
  width: ${cx(200)}; /* 避免显示过长文案 */
  text-align: center;
  font-weight: 500;
  font-size: 56px;
  color: ${cellFontColor};
  background-color: transparent;
`;

export const StyleDividerView = styled(View)`
  background-color: ${cellBg};
`;

export const StyleDivider = styled(View)`
  background-color: ${lineColor};
  height: ${StyleSheet.hairlineWidth};
`;

/**
 *  DatePicker
 */

export const StyledDatePicker = props => {
  return (
    <ThemeConsumer>
      {globalTheme => {
        const datePickerTheme = { ...props, theme: globalTheme };
        const fontColor = getTheme(datePickerTheme, 'popup.cellFontColor');
        let dateStyle;
        /* eslint-disable react/prop-types */
        if (props.style && props.style.backgroundColor) {
          dateStyle = props.style;
        } else if (props.style) {
          dateStyle = {
            ...props.style,
            backgroundColor: getTheme(datePickerTheme, 'popup.cellBg'),
          };
        } else {
          dateStyle = { backgroundColor: getTheme(datePickerTheme, 'popup.cellBg') };
        }
        return (
          <DatePicker
            {...props}
            pickerFontColor={fontColor}
            style={[
              !isIos && { height: cx(260), paddingTop: cx(32), paddingBottom: cx(32) },
              { paddingLeft: cx(30), paddingRight: cx(30) },
              dateStyle,
            ]}
          />
        );
      }}
    </ThemeConsumer>
  );
};

/**
 * Popup.List
 */

export const StyledFlatList = props => {
  return (
    <ThemeConsumer>
      {globalTheme => {
        const listTheme = { ...props, theme: globalTheme };
        return (
          <TYFlatList
            ItemSeparatorComponent={() => (
              <View style={{ backgroundColor: getTheme(listTheme, 'popup.cellBg') }}>
                <View
                  style={{
                    backgroundColor: getTheme(listTheme, 'popup.lineColor'),
                    marginLeft: 16,
                    marginRight: 16,
                    height: StyleSheet.hairlineWidth,
                  }}
                />
              </View>
            )}
            {...props}
          />
        );
      }}
    </ThemeConsumer>
  );
};

export const StyledIconFont = props => {
  const { color, ...rest } = props;
  return (
    <ThemeConsumer>
      {theme => {
        const propsWithTheme = { ...props, theme };
        return (
          <IconFont
            size={cx(28)}
            color={
              color ||
              getTheme(propsWithTheme, 'popup.checkboxColor', DEFAULT_LIST_THEME.checkboxColor)
            }
            {...rest}
          />
        );
      }}
    </ThemeConsumer>
  );
};
