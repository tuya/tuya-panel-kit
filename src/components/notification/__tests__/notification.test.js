import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Notification from '../index';
import { StyledNotificationContent } from '../styled';

// Shallow with New React Context API. Consumer not getting context: https://github.com/enzymejs/enzyme/issues/1636
jest.mock('../../../utils/theme', () => ({
  ThemeProvider: require.requireActual('../../../utils/theme').ThemeProvider,
  ThemeConsumer: props => props.children({ theme: { color: 'red' } }),
}));

describe('Notification Component', () => {
  jest.useFakeTimers();
  it('onLayout', () => {
    const wrapper = shallow(
      <Notification enableClose={false} show={true} theme={{ color: 'red' }} />
    );
    jest.runOnlyPendingTimers();
    // expect(wrapper).toMatchSnapshot();
    const target = wrapper.find('ThemeConsumer').dive().find(StyledNotificationContent);
    target.simulate('layout', { nativeEvent: { layout: { height: 44 } } });
  });
  it('componentWillReceiveProps(nextProps)', () => {
    const wrapper = shallow(
      <Notification enableClose={false} show={false} theme={{ color: 'red' }} />
    );
    jest.runOnlyPendingTimers();
    wrapper.setProps({ show: true });
  });
  it('basic render', () => {
    const component = renderer
      .create(
        Notification.show({
          message: '警告提示框',
          onClose: () => {
            Notification.hide();
          },
          theme: {
            successIcon: 'red',
            errorIcon: 'yellow',
            warningIcon: 'black',
          },
        })
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('render width update', () => {
    const component = renderer.create(
      Notification.show({
        message: '警告提示框',
        style: { width: 240, height: 36, borderRadius: 18 },
        enableClose: false,
        autoCloseTime: 1500,
        onPress: jest.fn(),
        theme: {
          iconColor: '#00f',
        },
      })
    );
    expect(component).toMatchSnapshot();
    component.update(Notification.hide());
    expect(component).toMatchSnapshot();
  });
});

describe('Notification', () => {
  it('basic render', () => {
    jest.useFakeTimers();
    const component = renderer.create(<Notification show={true} enableClose={false} />);
    jest.runOnlyPendingTimers();
    // expect(component).toMatchSnapshot();
    component.update(<Notification show={false} />);
    // expect(component).toMatchSnapshot();
  });

  it('basic render width enableClose', () => {
    const onClose = jest.fn();
    jest.useFakeTimers();
    const component = renderer.create(
      <Notification show={false} enableClose={false} autoCloseTime={1200} onClose={onClose} />
    );
    expect(component).toMatchSnapshot();
    component.update(
      <Notification show={true} enableClose={false} autoCloseTime={1200} onClose={onClose} />
    );
    jest.runAllTimers();
    expect(onClose).toHaveBeenCalled();
    component.unmount();
    expect(component).toMatchSnapshot();
  });
});
