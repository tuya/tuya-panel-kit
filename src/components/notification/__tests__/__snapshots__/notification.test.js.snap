// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Component basic render 1`] = `null`;

exports[`Notification Component render width update 1`] = `null`;

exports[`Notification Component render width update 2`] = `null`;

exports[`Notification basic render width enableClose 1`] = `null`;

exports[`Notification basic render width enableClose 2`] = `null`;
